### PR TITLE
Text Embeddings Feature

### PR DESCRIPTION
**Pull request by ihmaws**
Initial review of embeddings feature (@rstrahan to update PR title and description)

### CODE REVIEW

File: templates/master/ml-model/index.js [hunk 1]
```diff
@@ -0,0 +1,150 @@
...
+                                "Resource": [
+                                    {"Fn::Sub":"arn:aws:s3:::${BootstrapBucket}/${BootstrapPrefix}/ml_model/e5-large.tar.gz"}
+                                ]                      
+                            }
+                        ]
+                    }
+                }                
+            ],
+            "ManagedPolicyArns": [
+                "arn:aws:iam::aws:policy/AmazonSageMakerFullAccess"
```
Comment (tabdunabi) [inline templates/master/ml-model/index.js:146]: @rstrahan can we remove `AmazonSageMakerFullAccess`, and specify the required permissions only. This won't be approved by the Publisher/security review.

File: lambda/import/embeddings.js [hunk 2]
```diff
@@ -0,0 +1,55 @@
+const aws = require('aws-sdk');
```
Comment (ihmaws) [inline lambda/import/embeddings.js:1]: @rstrahan this file is identical to the one in es-proxy-layer. Can we add the esproxy layer to the import lambdas and call from there?

File: lambda/import/index.js [hunk 3]
```diff
@@ -1,22 +1,73 @@
...
+    var default_settings_param = process.env.DEFAULT_SETTINGS_PARAM;
+    var custom_settings_param = process.env.CUSTOM_SETTINGS_PARAM;
+    console.log("Getting Default QnABot settings from SSM Parameter Store: ", default_settings_param);
+    var default_settings = await get_parameter(default_settings_param);
+    console.log("Getting Custom QnABot settings from SSM Parameter Store: ", custom_settings_param);
+    var custom_settings = await get_parameter(custom_settings_param);
+    var settings = _.merge(default_settings, custom_settings);
+    console.log("Merged Settings: ", settings);
+    return settings;
+}
```
Comment (ihmaws) [inline lambda/import/index.js:64]: @rstrahan all of this functionality already exists in the qnabot-common-layer. Similar to the previous comment, can we add es-proxy-layer and qnabot-common-layer to the Import Lambdas?

**Review by ihmaws — COMMENTED** [review]
We can remove a decent amount of code duplication by leveraging the existing lambda layers

**Comment by rstrahan** [comment]
*   @rstrahan<https://github.com/rstrahan> this file is identical to the one in es-proxy-layer. Can we add the esproxy layer to the import lambdas and call from there?
Yes, I’d actually planned to do that, but more focused on getting it working fast..  Do you want to do that and send a PR?

File: lambda/import/index.js (hunk 3 above)
Comment (ihmaws) [reply lambda/import/index.js]: Done

File: lambda/import/embeddings.js (hunk 2 above)
Comment (ihmaws) [reply lambda/import/embeddings.js]: done

File: templates/master/ml-model/index.js (hunk 1 above)
Comment (ihmaws) [reply templates/master/ml-model/index.js]: Done. Will be merged in through a different PR.

File: templates/master/ml-model/index.js [hunk 4]
```diff
@@ -0,0 +1,150 @@
...
+var _=require('lodash');
+const util = require('../../util');
+
+module.exports={
+    "QnABotSMModelTarVersion": {
+        "Type": "Custom::S3Version",
+        "Properties": {
+          "ServiceToken": { "Fn::GetAtt": ["CFNLambda", "Arn"] },
+          "Bucket": { "Ref": "BootstrapBucket" },
+          "Key": { "Fn::Sub": "${BootstrapPrefix}/ml_model/e5-large.tar.gz" },
```
[the remark this answers is not in the document]
Comment (ihmaws) [reply templates/master/ml-model/index.js]: Done. Will be merged in through a different commit.

File: package-lock.json [hunk 5]
```diff
@@ -1,12 +1,12 @@
 {
   "name": "qnabot-on-aws",
-  "version": "5.2.6",
+  "version": "5.2.6-knn-exp-7",
```
[the remark this answers is not in the document]
Comment (ihmaws) [reply package-lock.json]: Merged branch with v5.2.7 and reset the version. Will update to v5.3.0 on launch.

**Comment by ihmaws** [comment]
Force pushed commit to rebase this branch onto develop.

**Review by tabdunabi — APPROVED** [review]
LGTM

**Pull request merged**